src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java

```diff
@@ -3,28 +3,38 @@
 import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorder;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorderObservable;
+import edu.kis.powp.jobs2d.macros.DriverCallRecorderObserver;
 
 /**
  * Line adapter - Job2dDriver with DrawPanelController object.
  */
-public class LineDriverAdapter implements Job2dDriver {
+public class LineDriverAdapter extends DriverCallRecorderObservable implements Job2dDriver {
 	private ILine line;
 	private int startX = 0, startY = 0;
 	private String name;
 
 	private DrawPanelController drawController;
 
-	public LineDriverAdapter(DrawPanelController drawController, ILine line, String name) {
+	public LineDriverAdapter(DrawPanelController drawController, ILine line, String name, DriverCallRecorder recorder) {
```

Comment:
If you would like to follow observer pattern, you should use DriverCallRecorderObserver type instead of DriverCallRecorder and DriverCallRecorder observer should be added externally (to avoid coupling objects within Observable)